src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -303,6 +303,15 @@ private TCommandResult ProcessResponse(ConnectionId connectionId, CommandMessage
                     throw new MongoCommandException(connectionId, message, _command, materializedDocument);
                 }
 
+                if (rawDocument.Contains("writeConcernError"))
+                {
+                    var materializedDocument = rawDocument.Materialize(binaryReaderSettings);
+                    var writeConcernError = materializedDocument["writeConcernError"].AsBsonDocument;
+                    var message = writeConcernError.AsBsonDocument.GetValue("errmsg", null)?.AsString;
+                    var writeConcernResult = new WriteConcernResult(materializedDocument);
+                    throw new MongoWriteConcernException(connectionId, message, writeConcernResult);
+                }
+
```

Comment:
This is the new place to check for writeConcernError.

It's at the same level where we check for { w : 0 } which we did just before this.

Reply:
It could of course. There are currently no unit tests for this entire class. What level of effort do you want to see put into unit testing this method and class?

Reply:
acknowledged